### PR TITLE
[SPARK-26347][mllib]MergeAggregate serialize and deserialize function use Bytebuffer to optimize

### PR DESCRIPTION
**Pull request by lcqzte10192193**
## What changes were proposed in this pull request?
MergeAggregate serialize and deserialize function can use ByteBuffer to optimize.

## How was this patch tested?
SummarizerSuite

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 1]
```diff
@@ -609,16 +611,16 @@ private[ml] object SummaryBuilderImpl extends Logging {
       = new SummarizerBuffer(requestedMetrics, requestedComputeMetrics)
 
     override def serialize(state: SummarizerBuffer): Array[Byte] = {
-      // TODO: Use ByteBuffer to optimize
-      val bos = new ByteArrayOutputStream()
+      // Use ByteBuffer
+      val bos = new ByteBufferOutputStream()
       val oos = new ObjectOutputStream(bos)
       oos.writeObject(state)
       bos.toByteArray
```
[the remark this answers is not in the document]
Comment (lcqzte10192193) [reply mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: ok,thanks

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 2]
```diff
@@ -18,16 +18,18 @@
 package org.apache.spark.ml.stat
 
 import java.io._
+import java.nio.ByteBuffer
 
 import org.apache.spark.annotation.{Experimental, Since}
 import org.apache.spark.internal.Logging
-import org.apache.spark.ml.linalg.{Vector, Vectors, VectorUDT}
+import org.apache.spark.ml.linalg.{Vector, VectorUDT, Vectors}
```
Comment (10110346) [inline mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: The order cannot be changed

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 3]
```diff
@@ -609,16 +611,16 @@ private[ml] object SummaryBuilderImpl extends Logging {
       = new SummarizerBuffer(requestedMetrics, requestedComputeMetrics)
 
     override def serialize(state: SummarizerBuffer): Array[Byte] = {
-      // TODO: Use ByteBuffer to optimize
-      val bos = new ByteArrayOutputStream()
+      // Use ByteBuffer
+      val bos = new ByteBufferOutputStream()
       val oos = new ObjectOutputStream(bos)
       oos.writeObject(state)
```
Comment (10110346) [inline mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: You must call `oos.close()` afer `oos.writeObject(state)`

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala (hunk 2 above)
Comment (lcqzte10192193) [reply mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala]: Thanks

**Comment by 10110346** [comment]
LGTM, except some comments.
My suggestion is to create a JIRA entry.

**Comment by 10110346** [comment]
cc @WeichenXu123

**Comment by mgaido91** [comment]
can you share some benchmarks in order to demonstrate the perf improvement?

**Comment by lcqzte10192193** [comment]
I've compared two ways,but local optimization test results are not obvious.

**Comment by mgaido91** [comment]
@lcqzte10192193 the point is: if there is no real use case benefiting from this, why should we do it? So if you can show that there is a use case which gets an improvement from this PR, it is ok, otherwise this may be useless.

**Comment by srowen** [comment]
Yeah, I don't see a reason this would be faster, and it doesn't sound like it is.

**Pull request closed without merging**